drivers/adc/ad7768-1/ad77681.c

```diff
@@ -1169,6 +1170,139 @@ int32_t ad77681_apply_gain(struct ad77681_dev *dev,
 	return ret;
 }
 
+/**
+ * Upload sequence for Programmamble FIR filter
+ * @param dev - The device structure.
+ * @param coeffs - Pointer to the desired filter coefficients array to be written
+ * @param num_coeffs - Count of active filter coeffs
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_programmable_filter(struct ad77681_dev *dev,
+				    const float *coeffs,
+				    uint8_t num_coeffs)
+{
+	uint8_t coeffs_buf[4], coeffs_index, check_back = 0, i, address;
+	uint32_t twait;
+	int32_t twos_complement, ret;
+	const uint8_t coeff_reg_length = 56;
+
+	/* Specific keys in the upload sequence */
+	const uint8_t key1 = 0xAC, key2 = 0x45, key3 = 0x55;
+	/* Scaling factor for all coefficients 2^22 */
+	const float coeff_scale_factor = (1 << 22);
+	/* Wait time in uS necessary to access the COEFF_CONTROL and */
+	/* COEFF_DATA registers. Twait = 512/MCLK */
+	twait = (uint32_t)(((512.0) / ((float)(dev->mclk))) * 1000.0) + 1;
+
+
+	/* Set Filter to FIR */
+	ret = ad77681_spi_write_mask(dev,
+				     AD77681_REG_DIGITAL_FILTER,
+				     AD77681_DIGI_FILTER_FILTER_MSK,
+				     AD77681_DIGI_FILTER_FILTER(AD77681_FIR));
+
+	/* Write the first access key to the ACCESS_KEY register */
+	ret |= ad77681_spi_write_mask(dev,
+				      AD77681_REG_ACCESS_KEY,
+				      AD77681_ACCESS_KEY_MSK,
+				      AD77681_ACCESS_KEY(key1));
```

Comment:
I think Darius ment:
```suggestion
	ret = ad77681_spi_write_mask(dev,
				     AD77681_REG_DIGITAL_FILTER,
				     AD77681_DIGI_FILTER_FILTER_MSK,
				     AD77681_DIGI_FILTER_FILTER(AD77681_FIR));
        if (ret < 0)
           return ret;
	/* Write the first access key to the ACCESS_KEY register */
	ret = ad77681_spi_write_mask(dev,
				      AD77681_REG_ACCESS_KEY,
				      AD77681_ACCESS_KEY_MSK,
				      AD77681_ACCESS_KEY(key1));
	if (ret < 0)
	     return ret;
``` 
       and so on.

Reply:
@buha @mchindri  apologies for missing it. Done implementing the suggested implementation.